libs/island-ui/core/src/lib/InputFileUpload/InputFileUpload.tsx

```diff
@@ -119,72 +151,42 @@ export const UploadedFile = ({
     }
   }, [file.status, file.size, defaultBackgroundColor])
 
-  const statusIcon = (status?: UploadFileStatus): IconTypes => {
-    switch (status) {
-      case 'error':
-        return 'close'
-      case 'done':
-        return doneIcon ?? 'close'
-      default:
-        return 'reload'
-    }
-  }
-
-  const kb = (bytes?: number) => {
-    return bytes ? Math.ceil(bytes / 1024) : ''
-  }
-
-  const truncateInMiddle = (str: string) => {
-    if (str.length > 40) {
-      const nrOfCharacters = width / 25
-      return `${str.slice(0, nrOfCharacters)}...${str.slice(-nrOfCharacters)}`
-    } else {
-      return str
-    }
-  }
-
   const isUploading =
     file.percent !== undefined &&
     file.percent < 100 &&
-    file.status === 'uploading'
+    file.status === FileUploadStatus.uploading
 
   return (
     <Box
-      ref={ref}
+      width="full"
+      position="relative"
       display="flex"
       flexDirection="row"
       alignItems="center"
       justifyContent="spaceBetween"
       borderRadius="large"
-      borderStyle={'solid'}
-      borderWidth={'standard'}
+      borderStyle="solid"
+      borderWidth="standard"
       borderColor={statusColor.border}
       background={statusColor.background}
       paddingX={2}
       paddingY={1}
-      width="full"
-      position="relative"
-      title={file.name}
-      aria-label={onOpenFile ? `Opna ${file.name}` : undefined}
       className={cn(styles.uploadedFile, {
         [styles.canOpenFiles]: onOpenFile,
       })}
-      onClick={(e) => {
-        e.stopPropagation()
-
-        if (onOpenFile) {
-          onOpenFile(file)
-        }
-      }}
+      onClick={(evt) => handleClick(evt, onOpenFile)}
+      title={file.name}
+      aria-label={onOpenFile ? `Opna ${file.name}` : undefined}
     >
-      <Text truncate fontWeight="semiBold">
+      <Text fontWeight="semiBold">
         <Box component="span" className={{ [styles.fileName]: onOpenFile }}>
-          {truncateInMiddle(file.name)}
-          {showFileSize && (
-            <Text as="span">{` (${file.size ? kb(file.size) : 0}KB)`}</Text>
+          {truncateMiddle(file.name)}
+          {file.size && (
```

Comment:
This will be false when `file.size === 0`, which we should allow in the `formatFileSize`

Reply:
Absolutely! Thanks. Fixed.